src/monitor.js

```diff
@@ -40,14 +40,46 @@ module.exports = (pluginContext) => {
   }
 
   let lastClip
-  return (env = {}) => {
-    if (isTransient()) { return Promise.resolve() }
-    const clip = getClip(env.ignoreImages)
-    if (!lastClip || lastClip.type !== clip.type || lastClip.raw !== clip.raw) {
-      lastClip = clip
-      const clipCollection = CappedClient.init(cwd, env)
-      return clipCollection.upsert(clip)
-    }
-    return Promise.resolve()
+  monitor = (env = {}) => {
+    return new Promise((resolve, reject) => {
+      if (isTransient()) {
+        resolve()
+      } else {
+        const clip = getClip(env.ignoreImages)
+        if (!lastClip || lastClip.type !== clip.type || lastClip.raw !== clip.raw) {
+          lastClip = clip
+          const clipCollection = CappedClient.init(cwd, env)
+          resolve(clipCollection.upsert(clip))
+        } else {
+          resolve()
+        }
+      }
+    })
   }
+
+  //  As there is a performance hit of clipboard on Linux platform,
+  //  Here we let Linux have more interval than other platform by default.
+  //  See also: https://github.com/tinytacoteam/zazu/issues/189
+  const DEFAULT_INTERVAL = (process.platform === 'linux' ? 3000 : 1000)
+
+  //  We use a loop here to provide user the ability to customize the interval.
+  //  it will keep looping unless got an exception.
+  start = (env = {}) => new Promise((resolve) => {
+      //  interval value check
+      let interval = parseInt(env.updateInterval, 10)
+      if (isNaN(interval)) {
+        interval = DEFAULT_INTERVAL
+      } else if (interval < 250) {
+        interval = 250
+      }
+      setTimeout(resolve, interval)
+    })
+    .then(() => monitor(env))
+    .then(() => start(env))
+    .catch((error) => {
+      console.log('error', error)
```

Comment:
I think the wrapping block will put this in the plugin debugger without you needing to catch/console/throw again.